app/commit.go

```diff
@@ -39,7 +39,9 @@ func (cs *CommitStore) CommitInfo() (weave.CommitID, error) {
 // need to think what concurrency we expect
 func (cs *CommitStore) Commit() (weave.CommitID, error) {
 	// flush deliver to store and discard check
-	cs.deliver.Write()
+	if err := cs.deliver.Write(); err != nil {
+		return weave.CommitID{}, err
+	}
```

Comment:
This is changing the functionality. I hope this is a fix :+1:

Reply:
yeah, same here :D

Reply:
Nice catch. I didn't even think about this silently ignoring error.

+1 for the linter